codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -1362,6 +1465,74 @@ let getTopicSelection = (eventFilters) => eventFilters->SingleOrMultiple.normali
             kind: system_config::EventKind::Params(vec![]),
             sighash: "0x50f7d27e90d1a5a38aeed4ceced2e8ec1ff185737aca96d15791b470d3f17363"
                 .to_string(),
+            field_selection: None,
+        })
+        .unwrap();
+
+        assert_eq!(
+            event_template,
+            EventTemplate {
+                name: "NewGravatar".to_string(),
+                params: vec![],
+                module_code: format!(
+                    r#"
+let sighash = "0x50f7d27e90d1a5a38aeed4ceced2e8ec1ff185737aca96d15791b470d3f17363"
+let topicCount = 1
+let name = "NewGravatar"
+let contractName = contractName
+
+@genType
+type eventArgs = unit
+@genType
+type block = Block.t
+@genType
+type transaction = Transaction.t
+
+@genType
+type event = Internal.genericEvent<eventArgs, block, transaction>
+@genType
+type loader<'loaderReturn> = Internal.genericLoader<Internal.genericLoaderArgs<event, loaderContext>, 'loaderReturn>
+@genType
+type handler<'loaderReturn> = Internal.genericHandler<Internal.genericHandlerArgs<event, handlerContext, 'loaderReturn>>
+@genType
+type contractRegister = Internal.genericContractRegister<Internal.genericContractRegisterArgs<event, contractRegistrations>>
+
+let paramsRawEventSchema = S.literal(%raw(`null`))->S.variant(_ => ())
+let blockSchema = Block.schema
+let transactionSchema = Transaction.schema
+
+let convertHyperSyncEventArgs = (Utils.magic: HyperSyncClient.Decoder.decodedEvent => eventArgs)
+
+let handlerRegister: HandlerTypes.Register.t = HandlerTypes.Register.make(
+  ~topic0=sighash->EvmTypes.Hex.fromStringUnsafe,
+  ~contractName,
+  ~eventName=name,
+)
+
+@genType
+type eventFilter = {{  }}
```

Comment:
Is something meant to be here?

Reply:
No, since the event doesn't have indexed params

Reply:
Why does it have a double brace though?

Reply:
This is to escape {} in rust. So in the generated code it'll be an empty object type.